src/exoticatechnologies/modifications/exotics/impl/AmmoCraftingSystem.kt

```diff
@@ -0,0 +1,447 @@
+                && weapon.ammoTracker.usesAmmo()
+                && weapon.ammoTracker.ammoPerSecond == 0f
+    }
+
+    private fun isMissileWeapon(weapon: WeaponAPI): Boolean {
+        return (
+                weapon.spec.mountType == WeaponAPI.WeaponType.MISSILE
+                        || weapon.type == WeaponAPI.WeaponType.MISSILE
+                        || weapon.spec.type == WeaponAPI.WeaponType.MISSILE
+                        || weapon.slot.weaponType == WeaponAPI.WeaponType.MISSILE
+                )
+    }
+
+    private fun isBallisticWeapon(weapon: WeaponAPI): Boolean {
+        return (
+                weapon.spec.mountType == WeaponAPI.WeaponType.BALLISTIC
+                        || weapon.type == WeaponAPI.WeaponType.BALLISTIC
+                        || weapon.spec.type == WeaponAPI.WeaponType.BALLISTIC
+                        || weapon.slot.weaponType == WeaponAPI.WeaponType.BALLISTIC
+                )
+    }
+
+    private fun reloadWeapon(weapon: WeaponAPI, member: FleetMemberAPI, mods: ShipModifications, exoticData: ExoticData) {
+        val reloadAmount = max(
+                1f, weapon.ammoTracker.maxAmmo * getPeriodReloadAmount(member = member, mods = mods, exoticData = exoticData)
+        ).roundToInt()  // 1 ammo or 10% of maxAmmo, whichever is higher - 10% of 8 is 0.8 so we want to reload at least 1 bullet in that case
+
+        weapon.ammoTracker.ammo = min(weapon.ammoTracker.maxAmmo, weapon.ammoTracker.ammo + reloadAmount)
+    }
+
+    inner class AmmoCreator(
+            ship: ShipAPI,
+            val member: FleetMemberAPI,
+            val mods: ShipModifications,
+            val exoticData: ExoticData
+    ) : MagicSubsystem(ship) {
+        private val affectedWeapons = ship.allWeapons.filter { weapon -> shouldAffectWeapon(weapon) }
+        private var systemActivated = AtomicBoolean(false)
+        private val timeCounter = IntervalUtil(
+                getPeriodDuration(member, mods, exoticData) - EPSILON, getPeriodDuration(member, mods, exoticData) + EPSILON
+        )
+        private val random = Random(System.nanoTime())
+
+        override fun getBaseActiveDuration(): Float = calculateSystemActivationDuration(member, mods, exoticData)
+
+        override fun getBaseCooldownDuration(): Float = calculateSystemCooldownDuration(member, mods, exoticData)
+
+        override fun shouldActivateAI(amount: Float): Boolean {
+            // This one is simple, if we have at least two weapons that aren't on max ammo - we should turn it on
+            var weaponsNotOnFullAmmo = 0
+            for (weapon in affectedWeapons) {
+                if (weapon.ammoTracker.ammo < weapon.ammoTracker.maxAmmo) {
+                    weaponsNotOnFullAmmo++
+                }
+                if (weaponsNotOnFullAmmo >= MIN_NONFULL_WEAPONS_NEEDED_FOR_ACTIVATION + 1) break
+            }
+
+            return weaponsNotOnFullAmmo >= MIN_NONFULL_WEAPONS_NEEDED_FOR_ACTIVATION
+        }
```

Comment:
consider changing this to activate only when we have an empty weapon, once the damage has been updated too

Reply:
Instead of activating when we have two non-full weapons, change to activate when one weapon is below 25% ammo.

Reply:
done

Reply:
resolving.